java-squid/src/main/java/org/sonar/java/se/SymbolicValue.java

```diff
@@ -91,7 +95,26 @@ public void computedFrom(List<SymbolicValue> symbolicValues) {
     }
 
     @Override
-    public ProgramState setConstraint(ProgramState programState, BooleanConstraint booleanConstraint) {
+    public List<ProgramState> setConstraint(ProgramState programState, NullConstraint nullConstraint) {
+      Object data = programState.constraints.get(this);
+      if (data instanceof NullConstraint) {
+        NullConstraint nc = (NullConstraint) data;
+        if ((NullConstraint.NULL.equals(nullConstraint) && NullConstraint.NOT_NULL.equals(nc)) ||
+          (NullConstraint.NULL.equals(nc) && NullConstraint.NOT_NULL.equals(nullConstraint))) {
+          // setting null where value is known to be non null or the contrary
+          return ImmutableList.of();
+        }
+      }
+      if (data == null || !data.equals(nullConstraint)) {
+        Map<SymbolicValue, Object> temp = Maps.newHashMap(programState.constraints);
+        temp.put(this, nullConstraint);
+        return ImmutableList.of(new ProgramState(programState.values, temp, programState.visitedPoints, programState.stack));
+      }
+      return ImmutableList.of(programState);
+    }
+
+    @Override
+    public List<ProgramState> setConstraint(ProgramState programState, BooleanConstraint booleanConstraint) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "setConstraint" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)